spec/requests/post_spec.rb

```diff
@@ -165,4 +165,70 @@
       end
     end
   end
+
+  describe 'DESTROY post' do
+    context 'when user is authenticated' do
+      let!(:users) { create_list(:user, 3) }
+      let!(:post) { create(:post, user: users[0]) }
+      let!(:post2) { create(:post, user: users[1]) }
+      let!(:comment) { create(:comment, commentable: post) }
+      let!(:like) { create(:like, likable: post) }
+      let!(:comment2) { create(:comment, commentable: post2) }
+      let!(:like2) { create(:like, likable: post2) }
```

Comment:
Let's do this whole logic from the post factory. You could do it using [Traits](https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#traits)

By adding a trait, you could implement logic in the post factory to create a post with likes.
By adding another trait, you could implement logic in the post factory to create a post with comments.
And by adding another trait, you could even implement logic  in the post factory to create a post with likes and comments.